tests/e2e/e2e_test.go

```diff
@@ -111,3 +109,257 @@ func (s *IntegrationTestSuite) TestBankTokenTransfer() {
 		)
 	})
 }
+
+func (s *IntegrationTestSuite) TestSendTokensFromNewGovAccount() {
+	chainAAPIEndpoint := fmt.Sprintf("http://%s", s.valResources[s.chainA.id][0].GetHostPort("1317/tcp"))
+	senderAddress, err := s.chainA.validators[0].keyInfo.GetAddress()
+	s.Require().NoError(err)
+	sender := senderAddress.String()
+	proposalCounter++
+	s.T().Logf("Proposal number: %d", proposalCounter)
+
+	s.fundCommunityPool(chainAAPIEndpoint, sender)
+
+	s.T().Logf("Submitting Legacy Gov Proposal: Community Spend Funding Gov Module")
+	s.submitLegacyProposalFundGovAccount(chainAAPIEndpoint, sender)
+	s.T().Logf("Depositing Legacy Gov Proposal: Community Spend Funding Gov Module")
+	s.depositGovProposal(chainAAPIEndpoint, sender, proposalCounter)
+	s.T().Logf("Voting Legacy Gov Proposal: Community Spend Funding Gov Module")
+	s.voteGovProposal(chainAAPIEndpoint, sender, proposalCounter, "yes", false)
+
+	initialGovBalance, err := getSpecificBalance(chainAAPIEndpoint, govModuleAddress, photonDenom)
+	s.Require().NoError(err)
+	proposalCounter++
+
+	s.T().Logf("Submitting Gov Proposal: Sending Tokens from Gov Module to Recipient")
+	s.submitNewGovProposal(chainAAPIEndpoint, sender, proposalCounter, "/root/.gaia/config/proposal_2.json")
+	s.T().Logf("Depositing Gov Proposal: Sending Tokens from Gov Module to Recipient")
+	s.depositGovProposal(chainAAPIEndpoint, sender, proposalCounter)
+	s.T().Logf("Voting Gov Proposal: Sending Tokens from Gov Module to Recipient")
+	s.voteGovProposal(chainAAPIEndpoint, sender, proposalCounter, "yes", false)
+	s.Require().Eventually(
+		func() bool {
+			sendGovAmount := sdk.NewInt64Coin(photonDenom, 10)
+			newGovBalance, err := getSpecificBalance(chainAAPIEndpoint, govModuleAddress, photonDenom)
+			s.Require().NoError(err)
+
+			recipientBalance, err := getSpecificBalance(chainAAPIEndpoint, govSendMsgRecipientAddress, photonDenom)
+			s.Require().NoError(err)
+			return newGovBalance.IsEqual(initialGovBalance.Sub(sendGovAmount)) && recipientBalance.Equal(initialGovBalance.Sub(newGovBalance))
+		},
+		15*time.Second,
+		5*time.Second,
+	)
+
+}
+
+func (s *IntegrationTestSuite) TestGovSoftwareUpgrade() {
+	chainAAPIEndpoint := fmt.Sprintf("http://%s", s.valResources[s.chainA.id][0].GetHostPort("1317/tcp"))
+	senderAddress, err := s.chainA.validators[0].keyInfo.GetAddress()
+	s.Require().NoError(err)
+	sender := senderAddress.String()
+	height := s.getLatestBlockHeight(s.chainA, 0)
+	proposalHeight := height + 35
```

Comment:
maybe a note here why 35 is significant?
did the number come about because it is enough blocks that we can assume the proposal has passed but not enough for the proposal to be executed?

Reply:
Yeah I can abstract this to a constant to explain, 35 ended up being the safest number of blocks to guarantee that everything could be processed in enough time.